### ISSUE TITLE
review path of storage of local offline drafts

### ISSUE THREAD

**Issue by tomholub** [FlowCrypt/flowcrypt-browser#3337]
This is after #3310 lands

I would like to review this in another issue and another PR - I'd rather have a more structured place where to save drafts.

For example a "localDrafts" array or object where all the drafts would be.

_Originally posted by @tomholub in https://github.com/FlowCrypt/flowcrypt-browser/pull/3310#discussion_r559498021_

**Comment by tomholub** [comment]
![image](https://user-images.githubusercontent.com/6306961/105221952-b6a9ec00-5b51-11eb-9cb9-253c2ff79e30.png)

**Comment by tomholub** [comment]
@limonte could you refactor this to put all the drafts as an array on a `localDrafts` storage property?

So instead of `await storageLocalSet({ [draftId]: { message: { raw, threadId } } });`

You would have `addDraftToLocalStorage` and `removeDraftFromLocalStorage` methods, maybe. Then these methods would be chaning a `localDrafts` storage field, which would be an array.

That way the drafts are more contained.

We may also need a migration of previous drafts to the new structure, here: https://github.com/FlowCrypt/flowcrypt-browser/blob/acacdd36a80c4142fcd3d868ee20dd68b7eb41e4/extension/js/background_page/migrations.ts#L40
